app/api/events.py

```diff
@@ -426,7 +426,13 @@ def before_get(self, args, kwargs):
         :param kwargs:
         :return:
         """
+        if 'id' in kwargs:
+            # condition to check if id is actually numerical identifier
+            if len(str(kwargs['id'])) >= 5:
```

Comment:
Why and how?

Reply:
Because when we are encountering the `event_id` transition from the frontend, it usually sends an identifier. But in case it's a complete integer, our api is being passed an `id` in the kwargs instead of `identifier` which messes up the `get_id` function.

So, if an id in the kwarg is a number and has a length of >=5 we can assume that it's actually an identifier being passed as an ID.
(at least that's what on my mind)

Reply:
The current mechanism is very brittle. The whole moving required params through kwargs is a bad idea and source of bugs. In future, the ID itself may be of more than 5 digits, then this will create a new bug. Can this be fixed in an earlier layer?

Reply:
@iamareebjamal I agree.
 
I had a plan, to make sure our api assign only non numeric identifiers. `1b3452` kind? This would ensure that we don't get the error at all but I am not sure if it would be a valid fix.

Reply:
This would be an OK workaround, but it won't solve errors for already generated identifiers

Reply:
@iamareebjamal Okay. I just want to know where should I look to figure out why is the api behaving this way and not taking only identifier kwargs.

Reply:
I don't know where it's making those conversions but most likely it's flask-json-restapi

Reply:
Understood, I shall have a look!